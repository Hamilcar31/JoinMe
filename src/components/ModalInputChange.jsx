import React, { useState } from 'react';
import {
  Modal, Text, View, StyleSheet,
} from 'react-native';
import { rgba } from 'polished';
import PropTypes from 'prop-types';
import InputBar from './InputBar';
import Button from './Button';


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: rgba(52, 52, 52, 0.8),
  },
  modal: {
    height: 170,
    alignItems: 'center',
    borderRadius: 10,
    padding: 10,
    marginLeft: 20,
    marginRight: 20,
    backgroundColor: 'white',
  },
  buttons: {
    flex: 1,
    flexDirection: 'row',
    marginTop: 10,
    marginBottom: 10,
  },
});

export default function ListItemModification(props) {
  const [modalVisible, setModalVisible] = useState(true);
  const [text, setText] = useState('');

  const {
    fieldName, fieldValue, callbackConfirm, callbackCancel,
  } = props;

  return (
    <View>
      <Modal
        animationType="fade"
        transparent
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}
      >
        <View style={styles.container}>
          <View style={styles.modal}>
            <View>
              <Text>{fieldName}</Text>
              <InputBar
                onChangeText={(value) => setText(value)}
                placeholder={fieldName}
                defaultValue={fieldValue}
              />
            </View>
            <View style={styles.buttons}>
              <Button
                title="apply"
                onPress={() => {
                  callbackConfirm(text);
                  setModalVisible(!modalVisible);
                }}
              />
              <Button
                title="cancel"
                style={{ backgroundColor: 'red' }}
                onPress={() => {
                  callbackCancel(text);
                  setModalVisible(!modalVisible);
                }}
              />
            </View>
          </View>
        </View>
      </Modal>
    </View>
  );
}

ListItemModification.propTypes = {
  fieldName: PropTypes.string,
  fieldValue: PropTypes.string,
  callbackConfirm: PropTypes.func,
  callbackCancel: PropTypes.func,
};

ListItemModification.defaultProps = {
  fieldName: '',
  fieldValue: '',
  callbackConfirm: () => {},
  callbackCancel: () => {},
};
