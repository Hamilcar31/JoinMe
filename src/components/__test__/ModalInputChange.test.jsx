import React from 'react';
import renderer from 'react-test-renderer';

import ModalInputChange from '../ModalInputChange';

describe('Modal input change component', () => {
  it('with defaults parameters', () => {
    // given
    const modalInputChange = <ModalInputChange />;

    // when
    const tree = renderer.create(modalInputChange).toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
