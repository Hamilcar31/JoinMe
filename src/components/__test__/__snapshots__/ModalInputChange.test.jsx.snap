// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal input change component with defaults parameters 1`] = `
<View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(52,52,52,0.8)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 170,
            "marginLeft": 20,
            "marginRight": 20,
            "padding": 10,
          }
        }
      >
        <View>
          <Text>
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderColor": "white",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 9,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 0,
                  },
                  "shadowOpacity": 0.32,
                  "shadowRadius": 5.46,
                  "width": 300,
                },
                Object {},
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder=""
              rejectResponderTermination={true}
              style={
                Object {
                  "height": "100%",
                  "width": "90%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2F74C5",
                "borderColor": "#2F74C5",
                "borderRadius": 20,
                "borderWidth": 1,
                "elevation": 5,
                "height": 42,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 160,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                  },
                  Object {},
                ]
              }
            >
              apply
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderColor": "#2F74C5",
                "borderRadius": 20,
                "borderWidth": 1,
                "elevation": 5,
                "height": 42,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 160,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                  },
                  Object {},
                ]
              }
            >
              cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
